src/ImageSharp/Memory/Allocators/ArrayPoolMemoryAllocator.Buffer{T}.cs

```diff
@@ -53,8 +53,13 @@ public override Span<T> GetSpan()
                 {
                     ThrowObjectDisposedException();
                 }
-
+#if SUPPORTS_CREATESPAN
+                ref byte r0 = ref MemoryMarshal.GetReference<byte>(this.Data);
+                return MemoryMarshal.CreateSpan(ref Unsafe.As<byte, T>(ref r0), this.length);
+#else
                 return MemoryMarshal.Cast<byte, T>(this.Data.AsSpan()).Slice(0, this.length);
```

Comment:
In theory here we know that the `byte` size is an exact multiple of our target length for the `T` span, so I was thinking we could be able to remove that final `Slice` call by pre-slicing the `Data` array. Something like:

```csharp
return MemoryMarshal.Cast<byte, T>(this.Data.AsSpan(0, this.length * sizeof(T)));
```

Tried out on sharplab ([here](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAJI8YUCAAcAymIBuvMDFwBuGvSasAchwFQFKtYykALbFAkAZbMHbc+A1dTUBmJqQYBhGgG8aDfwwSerLYGDAMwACeYQDaALoMACah2I4BgcGh4bw8DAA2MFwA5hjGaQF+ATEAUrwYAOKFYgoAFBiREjAQAGYtAGowYBjQKACUo3GV/sSu3LjY3eFSEthcADwAKgB8DI0Yy6vkm1sto1PpDADuxmLhGwwgDNz4q9hFMInnvtQXAcQA7AwALIwfDQSJAsy4Ux5FgePAYNZRMJoBjbNrGXi4FjJDDYFgAQVwBy4p1GLCkeQUMBadFRpSxLAKxVKo3K/gAvudzjU6o0uM0wG0Ol1egMhiNxpMfn9Zlx5osGCTjrsYPsVlxSMdTud0tdbmiHk8uC8uG8Pl9dX9ASCwVAIVCYXCEUjojBUeiGdjcfiiSTafTMdjmSVjAwAFQMXC8ABeopaG3GbPOXOoHKAA==)) but apparently the codegen is worse for some reason. So yeah I'm not really seeing a way to improve this without having access to the .NET 5 APIs 🤔

Reply:
Yep .NET 5 makes this easier.